src/avahi_publisher.cpp

```diff
@@ -150,6 +150,11 @@ void AvahiPublisher::start()
         return;
     is_running = true;
 
+    if (streams.size() == 0) {
+        log_debug("No streams found. Not publishing avahi services");
+        return;
```

Comment:
I did this because I was think in a future case where we have new devices being added in runtime. But for this implementation you are right. The correct is to change is_running after this check